include/jemalloc/internal/extent.h

```diff
@@ -465,6 +479,11 @@ extent_prof_alloc_time_set(extent_t *extent, nstime_t t) {
 	nstime_copy(&extent->e_alloc_time, &t);
 }
 
+static inline void
+extent_prof_recent_alloc_set(extent_t *extent, prof_recent_t *recent_alloc) {
```

Comment:
Agree that `reset()` makes things clearer. The usage case for `prof_recent_alloc_set(extent)` would slightly be different though: it's also called for initializing the `e_prof_recent_alloc` field when it turns out that we are not recording the allocation (i.e. in the `label_rollback` part in `prof_recent_alloc()`), so I'll not assert the field is non-`NULL` beforehand.

Reply:
I’m not following, we cannot assert recent_alloc non null here?

Reply:
Right: in the `label_rollback` part of `prof_recent_alloc()`, I'm setting `e_prof_recent_alloc` to `NULL`, but the logic isn't the case that I'm setting it from something non-`NULL`, because this is the first time I'm setting it.